scripts/installation/install_solc.py

```diff
@@ -1,69 +1,77 @@
 #!/usr/bin/env python3
 
+"""
+Download supported version of solidity compiler for the host operating system.
 
+This script depends on `nucypher.blockchain.eth.sol` submodule and `py-solc-x` library 
+but otherwise does not require installation of `nucypher` or its dependencies. 
 """
-This file is part of nucypher.
 
-nucypher is free software: you can redistribute it and/or modify
-it under the terms of the GNU Affero General Public License as published by
-the Free Software Foundation, either version 3 of the License, or
-(at your option) any later version.
 
-nucypher is distributed in the hope that it will be useful,
-but WITHOUT ANY WARRANTY; without even the implied warranty of
-MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-GNU Affero General Public License for more details.
+from os.path import dirname, abspath
 
-You should have received a copy of the GNU Affero General Public License
-along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
-"""
+import os
+import sys
+from pathlib import Path
 
+import nucypher
 
-"""
-Download supported version of solidity compiler for Linux based operating system. 
-This script depends on `nucypher.blockchain.eth.sol` submodule and `requests` library 
-but does not require installation of `nucypher`. 
-"""
 
-import platform
-import shutil
-from os.path import dirname, join, abspath
-from os import chmod
-import requests
+def get_solc_config_path() -> Path:
+    nucypher = Path('nucypher').absolute()
```

Comment:
NOTE: there has been some trepidation about using `absolute()` since it is undocumented - https://stackoverflow.com/questions/42513056/how-to-get-absolute-path-of-a-pathlib-path-object

See https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module

Reply:
Hmm - seems to be working...?  I think i'm comfortable using it for dev installs, keeping an eye out for any upstream API changes.